bim2sim/tasks/common/serialize_elements.py

```diff
@@ -0,0 +1,54 @@
+import pickle
+
+from bim2sim.tasks.base import ITask
+
+
+class SerializedElement:
+    def __init__(self, element):
+        self.guid = element.guid
+        self.element_type = element.__class__.__name__
+        for attr_name, attr_val in element.attributes.items():
+            setattr(self, attr_name, attr_val)
+
+    def __repr__(self):
+        return "<serialized %s (guid: '%s')>" % (
+            self.element_type, self.guid)
+
+
+class SerializeElements(ITask):
+    """Serialize element structure, run() method holds detailed information."""
+
+    reads = ('elements', 'space_boundaries', 'tz_elements')
+    touches = ('serialized_elements',)
+    single_use = True
+
+    def run(self, elements, space_boundaries, tz_elements):
+        """Make the element structure serializable.
+
+        As due to swigPy objects coming from IfcOpenShell we can't
+        directly serialize a whole bim2sim project or even the elements
+        structure with serializers like pickle. To still keep the element
+        structure information after a project run, we just copy the relevant
+        information like the attributes from the AttributeManager, guid and
+        type of the element to a simple SerializedElement instance and store it
+        with pickle.
+
+        Args:
+            elements: dict[guid: element] of bim2sim element structure
+            space_boundaries:
```

Comment:
please add documentation for space_boundaries and tz_elements

Reply:
solved